dask_expr/_collection.py

```diff
@@ -783,8 +783,7 @@ def shuffle(
         elif pd.api.types.is_list_like(on) and not is_dask_collection(on):
             on = list(on)
 
-        method = shuffle_method or get_default_shuffle_method()
-        if method == "p2p":
+        if shuffle_method or get_default_shuffle_method() == "p2p":
```

Comment:
```suggestion
        if (shuffle_method or get_default_shuffle_method()) == "p2p":
```


This is incorrect actually, == should bind stronger than or